src/Illuminate/Support/Arr.php

```diff
@@ -62,19 +63,24 @@ public static function build($array, callable $callback)
     /**
      * Collapse an array of arrays into a single array.
      *
-     * @param  array  $array
+     * @param  \Traversable|array  $array
      * @return array
      */
     public static function collapse($array)
     {
+        // Iterate on the array for performance.
```

Comment:
So basically, just by iterating less a dozen times over arrays of barely a thousand items, you can already count the milliseconds you are losing. And *this goes linearly* with the length of the array. Tens of thousands items? Tens of milliseconds.